-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-43916/38766

ERROR: Build errors found during clang-tidy run.
```
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02823/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/unique_ptr.h:1065:34: note: in instantiation of member function 'TypedValueMapVariable<reco::Candidate, reco::Candidate, StringObjectFunction<reco::Candidate>, int>::TypedValueMapVariable' requested here
--
PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h:125:9: error: no matching constructor for initialization of 'ValueMapVariable<Candidate, int, short>' [clang-diagnostic-error]
  125 |       : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, cc, skipNonExistingSrc),
      |         ^                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02823/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/unique_ptr.h:1065:34: note: in instantiation of member function 'TypedValueMapVariable<reco::Candidate, reco::Candidate, StringObjectFunction<reco::Candidate>, int, short>::TypedValueMapVariable' requested here
--
PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h:125:9: error: no matching constructor for initialization of 'ValueMapVariable<Candidate, int, unsigned char>' [clang-diagnostic-error]
  125 |       : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, cc, skipNonExistingSrc),
      |         ^                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02823/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/unique_ptr.h:1065:34: note: in instantiation of member function 'TypedValueMapVariable<reco::Candidate, reco::Candidate, StringObjectFunction<reco::Candidate>, int, unsigned char>::TypedValueMapVariable' requested here
--
PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h:125:9: error: no matching constructor for initialization of 'ValueMapVariable<Candidate, int, unsigned short>' [clang-diagnostic-error]
  125 |       : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, cc, skipNonExistingSrc),
      |         ^                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02823/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/unique_ptr.h:1065:34: note: in instantiation of member function 'TypedValueMapVariable<reco::Candidate, reco::Candidate, StringObjectFunction<reco::Candidate>, int, unsigned short>::TypedValueMapVariable' requested here
--
PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h:125:9: error: no matching constructor for initialization of 'ValueMapVariable<Candidate, unsigned int, unsigned int>' [clang-diagnostic-error]
  125 |       : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, cc, skipNonExistingSrc),
      |         ^                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02823/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/unique_ptr.h:1065:34: note: in instantiation of member function 'TypedValueMapVariable<reco::Candidate, reco::Candidate, StringObjectFunction<reco::Candidate>, unsigned int>::TypedValueMapVariable' requested here
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
+++ b/PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
@@ -122,7 +122,7 @@ public:
                         const edm::ParameterSet &cfg,
                         edm::ConsumesCollector &&cc,
                         bool skipNonExistingSrc = false)
-      : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, cc, skipNonExistingSrc),
+    : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, std::move(cc), skipNonExistingSrc),
         func_(cfg.getParameter<std::string>("expr"), true),
         precisionFunc_(cfg.existsAs<std::string>("precision") ? cfg.getParameter<std::string>("precision") : "23",
                        true) {}

--- a/PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
+++ b/PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
@@ -122,7 +122,7 @@ public:
                         const edm::ParameterSet &cfg,
                         edm::ConsumesCollector &&cc,
                         bool skipNonExistingSrc = false)
-    : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, std::move(cc), skipNonExistingSrc),
+      : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, std::move(cc), skipNonExistingSrc),
         func_(cfg.getParameter<std::string>("expr"), true),
         precisionFunc_(cfg.existsAs<std::string>("precision") ? cfg.getParameter<std::string>("precision") : "23",
                        true) {}

--- a/PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
+++ b/PhysicsTools/NanoAOD/interface/SimpleFlatTableProducer.h
@@ -82,20 +82,18 @@ public:
                     std::vector<edm::Ptr<ObjType>> selptrs,
                     nanoaod::FlatTable &out) const = 0;
 };
+
 template <typename ObjType, typename TIn, typename ValType = TIn>
-class ValueMapVariable : public ExtVariable<ObjType> {
+class ValueMapVariableBase : public ExtVariable<ObjType> {
 public:
-  ValueMapVariable(const std::string &aname,
-                   const edm::ParameterSet &cfg,
-                   edm::ConsumesCollector &&cc,
-                   bool skipNonExistingSrc = false)
+  ValueMapVariableBase(const std::string &aname,
+                       const edm::ParameterSet &cfg,
+                       edm::ConsumesCollector &&cc,
+                       bool skipNonExistingSrc = false)
       : ExtVariable<ObjType>(aname, cfg),
         skipNonExistingSrc_(skipNonExistingSrc),
         token_(cc.consumes<edm::ValueMap<TIn>>(cfg.getParameter<edm::InputTag>("src"))) {}
-  virtual ValType eval(const edm::Handle<edm::ValueMap<TIn>> &vmap, const edm::Ptr<ObjType> &op) const {
-    ValType val = (*vmap)[op];
-    return val;
-  }
+  virtual ValType eval(const edm::Handle<edm::ValueMap<TIn>> &vmap, const edm::Ptr<ObjType> &op) const = 0;
   void fill(const edm::Event &iEvent, std::vector<edm::Ptr<ObjType>> selptrs, nanoaod::FlatTable &out) const override {
     edm::Handle<edm::ValueMap<TIn>> vmap;
     iEvent.getByToken(token_, vmap);
@@ -115,14 +113,28 @@ protected:
   edm::EDGetTokenT<edm::ValueMap<TIn>> token_;
 };
 
+template <typename ObjType, typename TIn, typename ValType = TIn>
+class ValueMapVariable : public ValueMapVariableBase<ObjType, TIn, ValType> {
+public:
+  ValueMapVariable(const std::string &aname,
+                   const edm::ParameterSet &cfg,
+                   edm::ConsumesCollector &&cc,
+                   bool skipNonExistingSrc = false)
+      : ValueMapVariableBase<ObjType, TIn, ValType>(aname, cfg, std::move(cc), skipNonExistingSrc) {}
+  ValType eval(const edm::Handle<edm::ValueMap<TIn>> &vmap, const edm::Ptr<ObjType> &op) const {
+    ValType val = (*vmap)[op];
+    return val;
+  }
+};
+
 template <typename ObjType, typename TIn, typename StringFunctor, typename VarType, typename ValType = VarType>
-class TypedValueMapVariable : public ValueMapVariable<ObjType, VarType, ValType> {
+class TypedValueMapVariable : public ValueMapVariableBase<ObjType, TIn, ValType> {
 public:
   TypedValueMapVariable(const std::string &aname,
                         const edm::ParameterSet &cfg,
                         edm::ConsumesCollector &&cc,
                         bool skipNonExistingSrc = false)
-      : ValueMapVariable<ObjType, VarType, ValType>(aname, cfg, std::move(cc), skipNonExistingSrc),
+      : ValueMapVariableBase<ObjType, TIn, ValType>(aname, cfg, std::move(cc), skipNonExistingSrc),
         func_(cfg.getParameter<std::string>("expr"), true),
         precisionFunc_(cfg.existsAs<std::string>("precision") ? cfg.getParameter<std::string>("precision") : "23",
                        true) {}

